keps/sig-network/20180612-ipv4-ipv6-dual-stack.md

```diff
@@ -526,302 +517,382 @@ dualstack cluster can have this flag configured as
-   <secondary cidr entry>` is considered cluster default IP address family.
+5. The first CIDR entry of `--service-cluster-ip-range` is considered cluster
+   default IP address family.
 6. Attempting to allocate `ClusterIP` or `ClusterIPs` from an IP address family
    that is not configured on the cluster will result in an error.
 
-> The below discussion assumes that `Endpoint Slice and Endpoint Slice
-> Controller`  will allow dual stack endpoints, the existing endpoint
+> The below discussion assumes that `EndpointSlice and EndpointSlice
+> Controller`  will allow dual stack endpoints, the existing Endpoints
 > controller will remain single stack. That means enabling dual stack services
 > by having more than one entry in `--service-cluster-ip-range` without
-> enabling `Endpoint Slice` feature gate will result in validation failures
+> enabling `EndpointSlice` feature gate will result in validation failures
 > during startup.
 
 #### Type NodePort, LoadBalancer, ClusterIP
 
-In a dualstack cluster Services can be:
+In a dual stack cluster, Services can be:
 1. Single Stack: service IP are allocated (or reserved, if provided by user)
    from the first or the second CIDR (if configured) entry as configured using
    `--service-cluster-ip-range` flag, in this scenario services will have a
    single allocated `ClusterIP` (also set in `ClusterIPs`).
 2. Dual Stack (optional): service IP is allocated (or reserved, if provided by
    user) from both the first and the second CIDR (if configured) entries. In
-   this scenario services will have a dual assigned `ClusterIPs`. If the
+   this scenario services will have one or two assigned `ClusterIPs`. If the
    cluster is not configured for dualstack then  the resulting service will be
    created as a single stack and `ClusterIP/ClusterIPs` will be assigned from a
    single family.
 3. Dual Stack (required): if the cluster is not configured for dual stack then
    service creation will fail. If the cluster is configured for dual stack the
-   resulting service will carry two dual stack `ClusterIPs`.
+   resulting service will carry two `ClusterIPs`.
 
 The above is achieved using the following changes to Service api.
 1. Add a `spec.preferDualStack` optional boolean field.
-2. Add a `spec.ipFamilies[]` optional field with possible values of `IPv4`,
-   `IPv6` for each each entry or `nil` for the entire slice. Maximum entries in
-   this array is two and each entry must be `IPv4`or `IPv6` in any order. This
-   field identifies the expected IP families of a service. api-server will
-   reject the service creation request if the IP family is not supported by the
-   cluster. Further dependency and validation on this field is described below.
+2. Add a `spec.ipFamilies[]` optional field with possible values of `"IPv4"` or
+   `"IPv6"` for each each entry or `nil` for the entire slice. Maximum entries
+   in this array is two and each entry must be either `IPv4`or `IPv6`, in any
```

Comment:
It would be less text to just say "with possible values of `nil`, `["IPv4"]`, `["IPv6"]`, `["IPv4", "IPv6"]`, and `["IPv6", "IPv4"]`"...